packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+  return path.basename(fileName, '.json');
+}
+
+function isDevnet(network: string) {
+  return network.includes('dev');
+}
+
+function isTestnet(network: string) {
+  return ['mainnet', 'rinkeby', 'ropsten', 'kovan', 'goerli'].includes(network);
+}
+
+function getNewManifestLocation(oldName: string): string {
+  return isDevnet(oldName) ? oldName.replace('dev', 'unknown') : oldName;
+}
+
+async function writeJSONFile(location: string, data: unknown): Promise<void> {
+  await fs.writeFile(location, JSON.stringify(data, null, 2));
+}
+
+function transformProxies(proxies: LegacyProxies) {
+  return Object.keys(proxies).map(proxyName => ({
+    [path.basename(proxyName)]: proxies[proxyName].map(proxy => ({
+      address: proxy.address,
+      implementation: proxy.implementation,
+      admin: proxy.admin,
+    })),
+  }));
+}
+
+function updateManifest(oldManifest: LegacyManifest): ManifestData {
+  const proxyAdmin = oldManifest.proxyAdmin.address;
+
+  if (proxyAdmin === undefined) {
+    throw new Error('Legacy manifest does not have admin address');
+  }
+
+  if (Object.keys(oldManifest.solidityLibs).length === 0) {
+    throw new Error('Legacy manifest uses external Solidity libraries');
+  }
+
+  return {
+    manifestVersion: '3.1',
+    impls: transformImplementations(oldManifest.contracts),
+    admin: {
+      address: proxyAdmin,
+    },
+  };
+}
+
+function transformImplementations(contracts: LegacyContracts): Record<string, ImplDeployment> {
+  const impls: Record<string, ImplDeployment> = {};
+  for (const contractName in contracts) {
+    const contract = contracts[contractName];
+    if (contract.deployedBytecodeHash === undefined) {
+      continue;
+    } else {
+      impls[contract.deployedBytecodeHash] = transformImplementationItem(contract);
+    }
+  }
+  return impls;
+}
+
+function transformImplementationItem(contract: LegacyContract): ImplDeployment {
+  return {
+    address: contract.address || 'NO_ADDRESS_FOUND',
+    layout: transformLayout(contract),
+  };
+}
+
+function transformLayout(contract: LegacyContract): StorageLayout {
+  return {
+    storage: contract.storage?.map(transformStorageItem) || [],
+    types: transformTypes(contract.types || {}),
+  };
+}
+
+function transformStorageItem(storageItem: LegacyStorageItem): StorageItem {
+  return {
+    contract: storageItem.contract,
+    label: storageItem.label,
+    type: storageItem.type,
+    // TODO reconstruct path and line if sourcecode is available
+    src: storageItem.path,
+  };
+}
+
+function transformTypes(oldTypes: LegacyTypes): Record<string, TypeItem> {
+  const newTypes: Record<string, TypeItem> = {};
+  for (const _type in oldTypes) {
+    newTypes[_type] = {
+      label: oldTypes[_type].label,
+    };
+  }
+  return newTypes;
+}
+
+interface MigrationOutput {
+  migrations: Record<string, ManifestData>;
+  migrationData: unknown;
+}
+
+interface AddressWrapper {
+  address?: string;
+}
+
+interface LegacyManifest {
```

Comment:
Feel free to simplify this type by removing any properties we don't use, or using `unknown` in places where we don't use the value. Same for all the other types.

Reply:
This is a copypaste of the original types in the SDK.

Reply:
I'd personally follow YAGNI in this case, but it's not a strong opinion.